cast/src/lib.rs

```diff
@@ -443,6 +445,62 @@ where
         Ok(self.provider.get_gas_price().await?)
     }
 
+    pub async fn ls(&self) -> Result<String> {
+        // create a vector of addresses
+        let mut addrs: Vec<H160> = Vec::new();
+
+        // fetch the chain id
+        let chain_id = self.chain_id().await?;
+
+        // for each file in $HOME/.ethereum/keystore
+        if let Ok(home_dir) = std::env::var("HOME") {
+             let path = std::path::Path::new(&home_dir).join(".ethereum").join("keystore");
+             let entries = fs::read_dir(path)?;
+             for entry in entries {
+                 let entry = entry?;
+                 let path = entry.path();
+                 let file = std::io::BufReader::new(std::fs::File::open(&path)?);
+                 let mut value: serde_json::Value = serde_json::from_reader(file)?;
+
+                 let map = value.as_object_mut().unwrap();
+                 let addr = map.get_mut("address").unwrap().as_str().unwrap();
+                 let t = H160::from_str(addr).unwrap();
+                 addrs.push(t);
+             }
+        }
+
+        match Ledger::new(HDPath::LedgerLive(0), chain_id.as_u64()).await {
+            Ok(ledger) => {
+                for i in 0..6 {
+                    let path = HDPath::LedgerLive(i);
+                    let address = ledger.get_address_with_path(&path).await?;
+                    addrs.push(address)
+                }
+                for i in 0..6 {
+                    let path = HDPath::Legacy(i);
+                    let address = ledger.get_address_with_path(&path).await?;
+                    addrs.push(address)
+                }
+
+            }
+            Err(_) => {}
```

Comment:
should we print something here?

Reply:
`seth` doesn't print anything in this case, if its assumed that the output is safe to be consumed from a script then I don't think printing anything makes sense. I do think this command could use a `-v` flag